### PR TITLE
Added support for more features in the scripts

### PR DESCRIPTION
**Pull request by vpritamani-rubrik**
List of changes:

- Correlation of Backup Policies to Azure VM and VMSQL
- Backup Costs Added to AWS Script
- Added Tags to AWS Objects
- Provided in context logging on all scripts
- Added LTR to Azure SQL

### CODE REVIEW

File: CLOUD/Get-AWSSizingInfo.ps1 [hunk 1]
```diff
@@ -780,6 +918,55 @@ function getAWSData($cred) {
...
+  $endDate = (Get-Date).ToString("yyyy-MM-dd")
+  $timePeriod = @{
+      Start = $startDate
+      End = $endDate
+  }
+
+  $metrics = @("AmortizedCost", "BlendedCost", "NetAmortizedCost", "NetUnblendedCost", "NormalizedUsageAmount", "UnblendedCost", "UsageQuantity")
+
+  Write-Host "Getting Cost and Usage of Backup in Account: $awsAccountAlias" -ForegroundColor Green
+  Write-Progress -Activity 'Processing Cost and Usage of Backup:' -Status $plan.BackupPlanId -PercentComplete (0)
```
Comment (DamaniN) [inline CLOUD/Get-AWSSizingInfo.ps1:941]: This line causes an error when there are 0 backup plans. The error is:

`Write-Error: Cannot validate argument on parameter 'Status'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.`

**Review by DamaniN — CHANGES_REQUESTED** [review]
To include better errors when the scripts fail, change these lines in each script:

```pwsh
} catch{
  Write-Error $_
} finally{
  Stop-Transcript
}
```

To this:

```pwsh
} catch{
  Write-Error "An error occurred and the script has exited prematurely:"
  Write-Error $_
  Write-Error $_.ScriptStackTrace
} finally{
  Stop-Transcript
}
```

File: CLOUD/Get-AWSSizingInfo.ps1 (hunk 1 above)
Comment (vpritamani-rubrik) [reply CLOUD/Get-AWSSizingInfo.ps1]: @DamaniN resolved both of these errors

File: CLOUD/Get-AzureSizingInfo.ps1
[the remark this answers is not in the document]
Comment (DamaniN) [reply CLOUD/Get-AzureSizingInfo.ps1:1]: @vpritamani-rubrik  can you take a look?

**Review by DamaniN — CHANGES_REQUESTED** [review]
@vpritamani-rubrik  can you take a look at these changes?

**Comment by vpritamani-rubrik** [comment]
@DamaniN Implemented all of your feedback and corrected the scripts accordingly!

**Review by DamaniN — APPROVED** [review]
LGTM

**Pull request merged**